Mathlib/RingTheory/Ideal/Height.lean

```diff
@@ -0,0 +1,118 @@
+/-
+Copyright (c) 2025 Andrew Yang. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Jiedong Jiang, Jingting Wang, Andrew Yang
+-/
+import Mathlib.Order.KrullDimension
+import Mathlib.RingTheory.Ideal.MinimalPrime
+import Mathlib.RingTheory.PrimeSpectrum
+/-!
+# The Height of an Ideal
+
+In this file, we defined the height of a prime ideal and the height of an ideal.
+
+# Main definitions
+
+* `Ideal.primeHeight` : The height of a prime ideal $\mathfrak{p}$. We defined it as the length of
+the maximum chain of prime ideals whose maximal element is $\mathfrak{p}$ minus 1.
+
+* `Ideal.height` : The height of an ideal. We defined it as the infimum of the primeHeight of the
+minimal prime ideals containing I.
```

Comment:
```suggestion
* `Ideal.height` : The height of an ideal. We defined it as the infimum of the `primeHeight` of the
minimal prime ideals of I.
```
Or maybe "over". To me, "minimal prime ideals containing I" sounds like minimal prime ideals of `R` (over `0`) containing `I`.